agent/internal/fluent.go

```diff
@@ -0,0 +1,476 @@
+    record.log = record.log .. '\n'
+
+    return 2, timestamp, record
+end
+`
+	files = append(files,
+		archive.Item{
+			Path:     luaPath,
+			Type:     tar.TypeReg,
+			FileMode: 0444,
+			Content:  []byte(luaCode),
+		},
+	)
+
+	parserConfig := `
+[PARSER]
+  Name rank_id
+  Format regex
+  # Look for a rank ID from the beginning of the line (e.g., "[rank=0] xxx").
+  Regex ^\[rank=(?<rank_id>([0-9]+))\] (?<log>.*)
+
+[PARSER]
+  Name log_level
+  Format regex
+  # Look for a log level at the start of the line (e.g., "INFO: xxx").
+  Regex ^(?<level>(DEBUG|INFO|WARNING|ERROR|CRITICAL)): (?<log>.*)
+`
+
+	files = append(files,
+		archive.Item{
+			Path:     parserConfigPath,
+			Type:     tar.TypeReg,
+			FileMode: 0444,
+			Content:  []byte(parserConfig),
+		},
+	)
+
+	baseConfig := fmt.Sprintf(`
+[SERVICE]
+  # Flush every .05 seconds to reduce latency for users.
+  Flush .05
+  Parsers_File %s
+
+[INPUT]
+  Name forward
+`, parserConfigPath)
+
+	filterConfig := fmt.Sprintf(`
+# Attempt to parse the rank ID and log level out of output lines.
+[FILTER]
+  Name parser
+  Match *
+  Key_Name log
+  Parser rank_id
+  Reserve_Data true
+
+[FILTER]
+  Name parser
+  Match *
+  Key_Name log
+  Parser log_level
+  Reserve_Data true
+
+# Move around fields to create the desired shape of object.
+[FILTER]
+  Name modify
+  Match *
+  # Delete Docker's container information, which we don't want.
+  Remove container_id
+  Remove container_name
+  # Rename environment variables to normal names.
+  Rename DET_TRIAL_ID trial_id
+  Rename DET_CONTAINER_ID container_id
+
+  Add agent_id %s
+  Rename source stdtype
+
+# Apply the Lua code for miscellaneous field tweaking.
+[FILTER]
+  Name lua
+  Match *
+  Script %s
+  Call run
+`, opts.AgentID, luaPath)
+
+	// HACK: If a host resolves to both IPv4 and IPv6 addresses, Fluent Bit seems to only try IPv6 and
+	// fail if that connection doesn't work. IPv6 doesn't play well with Docker and many Linux
+	// distributions ship with an `/etc/hosts` that maps "localhost" to both 127.0.0.1 (IPv4) and [::1]
+	// (IPv6), so Fluent Bit will break when run in host mode. To avoid that, translate "localhost"
+	// diretcly into an IP address before passing it to Fluent Bit.
+	fluentMasterHost := opts.MasterHost
+	if fluentMasterHost == localhost {
+		fluentMasterHost = localhostIP
+	}
+
+	outputConfig := fmt.Sprintf(`
+# TMP: Output to stdout as well.
+[OUTPUT]
+  Name stdout
+  Match *
+
+[OUTPUT]
+  Name http
+  Match *
+  Host %s
+  Port %d
+  URI /trial_logs
+  Header_tag X-Fluent-Tag
+  Format json
+  Json_date_key timestamp
+  Json_date_format iso8601
+`, fluentMasterHost, opts.MasterPort)
+
+	if opts.Security.TLS.Enabled {
+		outputConfig += "  tls On\n"
+
+		if opts.Security.TLS.SkipVerify {
+			outputConfig += "  tls.verify Off\n"
+		}
+		if opts.Security.TLS.MasterCert != "" {
+			outputConfig += "  tls.ca_file " + masterCertPath + "\n"
+
+			certBytes, cErr := ioutil.ReadFile(opts.Security.TLS.MasterCert)
```

Comment:
comment: may be nice to add a "resolve" to the agent config and write a little package like the check package to do things like load the cert in and etc.. there's one forming in the master, just had to modify it to load the elastic cert so the master can send it to the agent.

Reply:
Hmm, interesting, but I don't feel like it's worth it yet. Maybe after this PR.